pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -280,7 +284,13 @@ func (t *Tracer) Init(gadgetCtx gadgets.GadgetContext) error {
 		return fmt.Errorf("installing tracer: %w", err)
 	}
 
+	if err := t.startGarbageCollector(gadgetCtx); err != nil {
+		t.Close()
+		return fmt.Errorf("starting garbage collector: %w", err)
+	}
+
 	t.ctx, t.cancel = gadgetcontext.WithTimeoutOrCancel(gadgetCtx.Context(), gadgetCtx.Timeout())
```

Comment:
Actually, I think the `newGarbageCollector` should receive the `gadgetCtx.Logger()` and also the `t.ctx` to wait on that instead of having its own channel to stop GC'ing. That would be the same mechanism we have in [toppers](https://github.com/inspektor-gadget/inspektor-gadget/blob/c83014e7b57c32120c32edc0451e6c0490a396dc/pkg/gadgets/top/tcp/tracer/tracer.go#L228).

That's indeed the aim of having such a context: go-routines within gadgets will all get notified if the user stopped the gadget (Ctrl+C) or if it was executed with `--timeout` and such time has passed.

Reply:
nice, that's simpler. Updated to use `t.ctx` Also changed it from a struct with methods to just a function `startGarbageCollector` since it no longer maintains state or needs a cleanup method.